tf/chunkparser.py

```diff
@@ -193,10 +245,65 @@ def parse_function(planes, probs, winner, q, plies_left):
 
         return (planes, probs, winner, q, plies_left)
 
-    def convert_v5_to_tuple(self, content):
+    def convert_v6_to_tuple(self, content):
+        """
+        Unpack a v6 binary record to 5-tuple (state, policy pi, result, q, m)
+
+        v6 struct format is (8356 bytes total):
+                                  size         1st byte index
+        uint32_t version;                               0
+        uint32_t input_format;                          4
+        float probabilities[1858];  7432 bytes          8
+        uint64_t planes[104];        832 bytes       7440
+        uint8_t castling_us_ooo;                     8272
+        uint8_t castling_us_oo;                      8273
+        uint8_t castling_them_ooo;                   8274
+        uint8_t castling_them_oo;                    8275
+        uint8_t side_to_move_or_enpassant;           8276
+        uint8_t rule50_count;                        8277
+        // Bitfield with the following allocation:
+        //  bit 7: side to move (input type 3)
+        //  bit 6: position marked for deletion by the rescorer (never set by lc0)
+        //  bit 5: game adjudicated (v6)
+        //  bit 4: max game length exceeded (v6)
+        //  bit 3: best_q is for proven best move (v6)
+        //  bit 2: transpose transform (input type 3)
+        //  bit 1: mirror transform (input type 3)
+        //  bit 0: flip transform (input type 3)
+        uint8_t invariance_info;                     8278
+        uint8_t dummy;                               8279
+        float root_q;                                8280
+        float best_q;                                8284
+        float root_d;                                8288
+        float best_d;                                8292
+        float root_m;      // In plies.              8296
+        float best_m;      // In plies.              8300
+        float plies_left;                            8304
+        float result_q;                              8308
+        float result_d;                              8312
+        float played_q;                              8316
+        float played_d;                              8320
+        float played_m;                              8324
+        // The folowing may be NaN if not found in cache.
+        float orig_q;      // For value repair.      8328
+        float orig_d;                                8332
+        float orig_m;                                8336
+        uint32_t visits;                             8340
+        // Indices in the probabilities array.
+        uint16_t played_idx;                         8344
+        uint16_t best_idx;                           8346
+        uint64_t reserved;                           8348
         """
-        Unpack a v5 binary record to 5-tuple (state, policy pi, result, q, m)
 
+        version = content[0:4]
+        if version == V6_VERSION:
```

Comment:
If you leave dummy as a signed value - we can skip this conditional and just unconditionally unpack as v6_struct.

Reply:
fixed

Reply:
The conditional is still here.

Reply:
ugh